pkg/sql/schema_changer.go

```diff
@@ -1642,7 +1642,7 @@ func (sc *SchemaChanger) done(ctx context.Context) error {
 							return err
 						}
 					}
-					scTable.RowLevelTTL = nil
+					scTable.RowLevelTTL = modify.RowLevelTTL()
```

Comment:
Do not assume `modify.RowLevelTTL()` is nil even though it will be based on logic to use direction=ADD.